components/automate-gateway/gateway/services.go

```diff
@@ -874,12 +874,15 @@ func (s *Server) authRequest(r *http.Request, resource, action string) (context.
 
 	projects := auth_context.ProjectsFromMetadata(md)
 
-	authzResp, err := s.authorizer.IsAuthorized(ctx, subjects, resource, action, projects)
+	newCtx, authorized, err := s.authorizer.IsAuthorized(ctx, subjects, resource, action, projects)
 	if err != nil {
+		// any error is handled as "permission denied" at the call sites, with the error text returned
 		return nil, errors.Wrap(err, "authz-service error")
 	}
-
-	// err is nil if authorization succeeded
-	// Note: if we need all the auth info, use auth_context.NewOutgoingContext
-	return auth_context.NewOutgoingProjectsContext(authzResp.Ctx()), authzResp.Err()
+	if authorized {
+		// Note: if we need all the auth info, use auth_context.NewOutgoingContext
```

Comment:
this is like, a "someday if we want" note?

Reply:
( i know you didn't write it, just wonderin )

Reply:
I think it's "if we wanted X, we should do Y here", so, yeah, your interpretation.